src/pyFAI/detectors/orientation.py

```diff
@@ -0,0 +1,68 @@
+# !/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+#    Project: Azimuthal integration
+#             https://github.com/silx-kit/pyFAI
+#
+#    Copyright (C) 2023-2023 European Synchrotron Radiation Facility, Grenoble, France
+#
+#    Principal author:       Jérôme Kieffer (Jerome.Kieffer@ESRF.eu)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+"""Orientation of detectors:
+
+This module contains mostly an enum with the associated documentation
+For orientation description, see http://sylvana.net/jpegcrop/exif_orientation.html
+"""
+
+
+__author__ = "Jérôme Kieffer"
+__contact__ = "Jerome.Kieffer@ESRF.eu"
+__license__ = "MIT"
+__copyright__ = "European Synchrotron Radiation Facility, Grenoble, France"
+__date__ = "21/11/2023"
+__status__ = "stable"
+
+from enum import IntEnum
+
+class Orientation(IntEnum):
+    """Names inspired from
+    https://learn.microsoft.com/en-us/uwp/api/windows.storage.fileproperties.photoorientation
+    """
+    Unspecified = 0
+    Normal = 1
+    FlipHorizontal = 2
+    Rotate180 = 3
+    FlipVertical = 4
+    Transpose = 5
+    Rotate270 = 6
+    Transverse = 7
+    Rotate90 = 8
+
+Orientation(0).__doc__ = "An orientation flag is not set."
+Orientation(1).__doc__ = "Camera default. Origin at the top left of the image when looking at the sample."
```

Comment:
So origin at top right when looking from the sample ?

Reply:
This enum is the default one used in EXIF, for visible light cameras.

In pyFAI, the origin is at the bottom left when looking from the sample. Thus the orientation should be flipped-ud for the origin at the bottom and flipped-rl to change the direction of sight. This makes the default orientation used in pyFAI to `3` or `Rotate180`.

I just got the answer from Dectris that their detector have to be seen from the sample and not from the other side.

At this stage, orientation is just shipped around and used nowhere ...

Reply:
Ok.

When speaking about images, looking at the detector from the sample makes sense. Looking at the sample from the detector is a bit harder to figure out.

Reply:
Looking at the sample is the standard when considering camera ... outside the X-ray scattering world.

https://www.researchgate.net/profile/Romain-Neuville/publication/330564129/figure/fig2/AS:718599043952648@1548338771876/The-virtual-camera-settings-camera-position-camera-orientation-focal-length-and.jpg

Reply:
https://www.researchgate.net/profile/Sunil-Kumar-Kopparapu/publication/220632389/figure/fig2/AS:883184350789637@1587578966909/Orientation-angles-rotation-about-the-camera-x-y-z-axes-are-referred-to-as-pitch.pbm

Reply:
Yeah. After thinking again about it this morning, this is indeed just a personal feeling.